File: client/clojure/src/mednet_edc_api/specs/site.clj
(ns mednet-edc-api.specs.site
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def site-data
  {
   (ds/opt :studyKey) string?
   (ds/opt :siteId) int?
   (ds/opt :siteName) string?
   (ds/opt :siteEnrollmentStatus) string?
   (ds/opt :dateCreated) string?
   (ds/opt :dateModified) string?
   })

(def site-spec
  (ds/spec
    {:name ::site
     :spec site-data}))

File: src/App/DTO/Site.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

class Site
{
    /**
     * Unique study key
     * @DTA\Data(field="studyKey", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $study_key = null;

    /**
     * Unique site ID
     * @DTA\Data(field="siteId", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     */
    public ?int $site_id = null;

    /**
     * Name of the site
     * @DTA\Data(field="siteName", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $site_name = null;

    /**
     * Enrollment status of the site
     * @DTA\Data(field="siteEnrollmentStatus", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $site_enrollment_status = null;

    /**
     * Date when this site was created
     * @DTA\Data(field="dateCreated", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $date_created = null;

    /**
     * Date when this site was last modified
     * @DTA\Data(field="dateModified", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $date_modified = null;

}
